src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo).append("\n");
+        LocalDate from = null;
+        LocalDate to = null;
+
+        try {
+            from = LocalDate.parse(dateFrom, formatter);
+            to = LocalDate.parse(dateTo, formatter);
+        } catch (DateTimeParseException e) {
+            System.out.println("Error parsing from & to Dates after initialize"
+                    + dateFrom + " "
+                    + dateTo + ". Error: "
+                    + e);
+        }
+
+        List<String> employeeNameByKey = new ArrayList<>(Arrays.stream(names).toList());
+        List<Integer> employeeSumByKey
+                = new ArrayList<>(Collections.nCopies(names.length, 0));
+
+        for (int i = 0; i < data.length; i++) {
+            String[] splittedString = data[i].split(" ");
+            LocalDate paymentDate = null;
+
+            try {
+                paymentDate = LocalDate.parse(splittedString[0], formatter);
+            } catch (DateTimeParseException e) {
+                System.out.println("Error parsing paymentDate by name from splittedString. "
+                        + "Parsed data: "
+                        + splittedString[0]
+                        + ". Error: "
+                        + e);
```

Comment:
Similar to the previous comment, the error message here could be more informative. Specify which part of the data failed to parse and include the problematic string in the message.